internal/config/config.go

```diff
@@ -155,15 +160,6 @@ func (c *Config) Validate() error {
 	if c.Parameters.VaultRoleName == "" {
 		return errors.Errorf("missing 'roleName' in SecretProviderClass definition")
 	}
-	certificatesConfigured := c.Parameters.TLSConfig.CertificatesConfigured()
-	if c.Parameters.TLSConfig.VaultSkipTLSVerify && certificatesConfigured {
-		return errors.New("both vaultSkipTLSVerify and TLS configuration are set")
-	}
-	if !certificatesConfigured &&
-		!c.Parameters.TLSConfig.VaultSkipTLSVerify &&
-		strings.HasPrefix(c.Parameters.VaultAddress, "https") {
-		return errors.New("no TLS configuration and vaultSkipTLSVerify is false but vault address scheme is https")
```

Comment:
Since we're not doing validation here, does it mean that we rely on the first client request failure to indicate that TLS configuration was not set up properly?

Reply:
We now rely on the validation logic from the Vault API package's `ConfigureTLS` function instead. Previously we didn't use that function all the time in order to continue supporting the `vaultCAPEM` option to pass the CA as a string in the SecretProviderClass. But it didn't seem worth continuing to support that given it meant maintaining an awful lot of code to roll our own HTTP client. And it also seems like a weird deployment pattern to paste your CA certificate into a secret provider class config anyway. That switch is the main reason this PR manages to net delete lines despite adding a new feature 😄